include/bdldl/bdldl.h

```diff
@@ -62,41 +80,45 @@ void sparseLDL_solve(const struct ldl_Solver *solver, const ldl_vector *b,
  * @param[in] num_states State dimension of every stage. Size `N + 1`
  * @param[in] num_inputs State dimension of every stage. Size `N + 1`
  * @param[in] num_constraints Constraint dimension of every stage. Size `N + 1`
- * @param[in] num_horizon Horizon length `N`
+ * @param[in] num_horizon Number of intermediate states `N`. For example, if N = 2, then
+ * x0(initial state) and x1 are intermediate states, x2 is the final state.
  * @param[in] use_diagonal_costs Use diagonal costs. If true, only memory for diagonal
  * entries is allocated; otherwise, memory for dense Q and R matrices is allocated.
  * @return size_t Required memory size
  */
-size_t sparseLDL_getRequiredMemorySize(const int *num_states, const int *num_inputs,
-                                       const int *num_constraints, int num_horizon,
-                                       ldl_bool use_diagonal_costs);
+size_t bdldl_getRequiredMemorySize(const int *num_states, const int *num_inputs,
+                                   const int *num_constraints, int num_horizon,
+                                   ldl_bool use_diagonal_costs);
 
 /**
- * (Allocate &) map the solver.
+ * (Allocate required memory &) map the solver to the given memory(or the internally
+ * allocated memory).
  *
  * @param[in] num_states State dimension of every stage. Size `N + 1`
  * @param[in] num_inputs State dimension of every stage. Size `N + 1`
  * @param[in] num_constraints Constraint dimension of every stage. Size `N + 1`
- * @param[in] num_horizon Horizon length `N`
+ * @param[in] num_horizon Number of intermediate states `N`. See \ref
+ * bdldl_getRequiredMemorySize for details.
  * @param[in] use_diagonal_costs Use diagonal costs. If true, only memory for diagonal
  * entries is allocated; otherwise, memory for dense Q and R matrices is allocated.
  * @param[in] memory If NULL, required memory is allocated internally. User should call
- * \ref sparseLDL_freeSolver to free the allocated memory explicitly. If a valid memory
+ * \ref bdldl_freeSolver to free the allocated memory explicitly. If a valid memory
  * address, solver is mapped to this pre-allocated memory space. Users should manage the
  * memory on their own.
  * @return struct ldl_Solver* Pointer to the solver
  *
  */
-struct ldl_Solver *sparseLDL_newSolver(const int *num_states, const int *num_inputs,
-                                       const int *num_constraints, int num_horizon,
-                                       ldl_bool use_diagonal_costs, void *memory);
+struct ldl_Solver *bdldl_newSolver(const int *num_states, const int *num_inputs,
```

Comment:
Also need error-handling in this function. That's usually best done via an `int*` or `bdldl_status*`, and remember to mention that in the docstring, and that the function will return a `NULL` if the error code is nonzero.